internal/provisioner/plain/controllers/bundledeployment_controller.go

```diff
@@ -370,14 +371,13 @@ func (r *BundleDeploymentReconciler) loadBundle(ctx context.Context, bundle *ruk
 		return nil, fmt.Errorf("load bundle: %v", err)
 	}
 
-	objects, err := getObjects(bundleFS)
+	plainV0, err := bundleLib.LoadPlainV0(bundleFS)
 	if err != nil {
 		return nil, fmt.Errorf("read bundle objects from bundle: %v", err)
 	}
 
-	objs := make([]client.Object, 0, len(objects))
-	for _, obj := range objects {
-		obj := obj
+	objs := make([]client.Object, 0, len(plainV0.Objects))
+	for _, obj := range plainV0.Objects {
```

Comment:
Any reason we can't/shouldn't modify the `plainV0.Objects` slice directly? It would use less memory if we modified the objects in place.

Reply:
at some point i was doing the type assert here, this was just an artifact of that. good catch!